### PR TITLE
SlackAdapter can now parse select options #3537

### PR DESCRIPTION
**Pull request by marcusbooyah**
I also added the `callback_id` property to the `SlackPayload` class.

### CODE REVIEW

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/SlackHelper.cs [hunk 1]
```diff
@@ -175,7 +175,21 @@ public static Activity PayloadToActivity(SlackPayload slackPayload)
             if (slackPayload.Actions != null && (slackPayload.Type == "block_actions" || slackPayload.Type == "interactive_message"))
             {
                 activity.Type = ActivityTypes.Message;
-                activity.Text = slackPayload.Actions[0].Value;
+
+                switch (slackPayload.Actions[0].Type)
```
[the remark this answers is not in the document]
Comment (garypretty) [reply libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/SlackHelper.cs:179]: @marcusbooyah can you please address the above comment and add a check along side the slackPayload.Actions != null to check for slackPayload.Actions.Any(). Thanks.

**Comment by garypretty** [comment]
@marcusbooyah Thank you very much for your contribution here.  I am going to hold off on merging this, even once the requested change has been made as I am currently reviewing the underlying client being used by the adapter and if we might swap this out for another.  This is because we have a growing number of Slack model classes in our adapter, in addition to those model classes within the client currently being used - this is because the client does not appear to be kept up to date with the change within the Slack API.  I am exploring an alternative that might include the changes required for selection options etc.

I will report back once my analysis has been completed.

**Comment by garypretty** [comment]
Based on current thinking work to merge this PR will happen over the next 2 weeks and the PR will make it into R10.  We have an overlapping PR #3760 which I will work to consolidate.

**Comment by sandeepna2206** [comment]
Thanks for the update @garypretty . Looking forward for R10.

File: tests/Adapters/Microsoft.Bot.Builder.Adapters.Slack.Tests/SlackAdapterTests.cs [hunk 2]
```diff
@@ -650,8 +650,12 @@ public async Task ProcessAsyncShouldSucceedOnSlashCommand()
             Assert.True(callback);
         }
 
-        [Fact]
-        public async Task ProcessAsyncShouldSucceedOnInteractiveMessage()
+        [Theory]
+        [InlineData(@"/Files/InteractiveMessageBodyButton.txt")]
```
Comment (cleemullins) [inline tests/Adapters/Microsoft.Bot.Builder.Adapters.Slack.Tests/SlackAdapterTests.cs:654]: These won't work across PC / Mac / Linux due to the "/" vs "\" problem.

File: tests/Adapters/Microsoft.Bot.Builder.Adapters.Slack.Tests/SlackHelperTests.cs [hunk 3]
```diff
@@ -207,6 +207,20 @@ public void PayloadToActivityShouldThrowArgumentNullException()
             Assert.Throws<ArgumentNullException>(() => SlackHelper.PayloadToActivity(null));
         }
 
+        [Theory]
+        [InlineData(@"/Files/InteractiveMessageBodyButton.txt", "recommend")]
```
Comment (cleemullins) [inline tests/Adapters/Microsoft.Bot.Builder.Adapters.Slack.Tests/SlackHelperTests.cs:211]: Same "/" vs "\"

File: tests/Adapters/Microsoft.Bot.Builder.Adapters.Slack.Tests/SlackHelperTests.cs [hunk 4]
```diff
@@ -207,6 +207,20 @@ public void PayloadToActivityShouldThrowArgumentNullException()
             Assert.Throws<ArgumentNullException>(() => SlackHelper.PayloadToActivity(null));
         }
 
+        [Theory]
```
Comment (cleemullins) [inline tests/Adapters/Microsoft.Bot.Builder.Adapters.Slack.Tests/SlackHelperTests.cs:210]: Love seeing theory tests. Nice work.

**Comment by garypretty** [comment]
We have now merged a PR (just awaiting some resolution of failing tests) which implements some changes to the Slack adapter, including those within this PR. @marcusbooyah , thank you very much for your contribution here.

The changes in the PR, which consolidates a number of changes, also removes the tight dependency on the Slack client that was being used. Now, a developer can install and use any client they wish directly from NuGet.

You can see the PR at #4046. Please let me know if you have any questions.

I am going to close out this PR and you should now see the updated Slack adapter in the nightly builds.

**Pull request closed without merging**